app/uk/gov/hmrc/gform/sharedmodel/formtemplate/AuthConfig.scala

```diff
@@ -29,7 +29,8 @@ object EEITTAuthConfig {
 
 case class EnrolmentAuth(
   serviceId: ServiceId,
-  enrolmentCheck: EnrolmentCheck
+  enrolmentCheck: EnrolmentCheck,
+  legacyFcEnrolmentVerifier: Option[LegacyFcEnrolmentVerifier]
```

Comment:
`LegacyFcEnrolmentVerifier` should not be part of `EnrolmentAuth`. Spec says that `LegacyFcEnrolmentVerifier` is valid only when Enrolment section exists. 

Existence of Enrolment section is witnessed by instance of `RequireEnrolment`, so we need to put `LegacyFcEnrolmentVerifier` into `RequireEnrolment`. This is what I understand as TDD (Type Driven Development).

Reply:
By using `Option` for `legacyFcEnrolmentVerifier` we are losing readability in a sense that we are using `None` as its value here and there and from a look at such a code it is hard to guess what type of `None` really is.

We can increase readability by introducing specific type which will be isomorphic with `Option[LegacyFcEnrolmentVerifier]` to alleviate this problem. 

We've used this same technique to have better representation of `Boolean` here [ObligationsAction.scala](https://github.com/hmrc/gform-frontend/blob/master/app/uk/gov/hmrc/gform/models/gform/ObligationsAction.scala#L19-L21)